src/main/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketClientFactoryImpl.java

```diff
@@ -298,23 +298,22 @@ private void addCredentials(Request.Builder builder) {
         Request.Builder requestBuilder = new Request.Builder().url(url);
         addCredentials(requestBuilder);
 
-        try {
-            Response response = okHttpClient.newCall(requestBuilder.build()).execute();
+        try (Response response = okHttpClient.newCall(requestBuilder.build()).execute()) {
             int responseCode = response.code();
 
-            try (ResponseBody body = response.body()) {
-                if (response.isSuccessful()) {
-                    if (body == null) {
-                        log.debug("Bitbucket - No content in response");
-                        throw new NoContentException(
-                                "Remote side did not send a response body", responseCode);
-                    }
-                    log.trace("Bitbucket - call successful");
-                    return new BitbucketResponse<>(
-                            response.headers().toMultimap(), reader.readObject(body.byteStream()));
+            ResponseBody body = response.body();
+            if (response.isSuccessful()) {
+                if (body == null) {
+                    log.debug("Bitbucket - No content in response");
+                    throw new NoContentException(
+                            "Remote side did not send a response body", responseCode);
                 }
-                handleError(responseCode, body == null ? null : body.string());
+                log.trace("Bitbucket - call successful");
```

Comment:
I think at the very least we should add the `url` to the trace message. `Bitbucket - call successful` message is not quite useful.